packages/cache/src/chunkPromise.ts

```diff
@@ -0,0 +1,16 @@
+type PromiseFn = () => Promise<unknown>;
+
+export async function chunkPromise(promises: (Promise<unknown> | PromiseFn)[], limit = 5) {
+  const chunk: (Promise<unknown> | PromiseFn)[] = [];
+  const lastPromise = promises[promises.length - 1];
+
+  for (const promise of promises) {
```

Comment:
Could this be simplified into something like this:

```
for (let i = 0; i < promises.length; i += limit) {
  await Promise.all(
    promises.slice(i, limit).map((p) => (typeof p === "function" ? p() : p)
  );
}
```

Reply:
Going to try!